src/CoreLib/assembly.fs

```diff
@@ -227,7 +227,7 @@ and [<AllowNullLiteral>]
             x.Pack( ms )  
             // x.Hash <- HashByteArrayWithLength( ms.GetBufferPosLength()  )
             let (ms, pos, len) = ms.GetBufferPosLength()
-            x.Hash <- ms.ComputeSHA256(int64 pos, int64 len)
+            x.Hash <- ms.ComputeDefaultHash(int64 pos, int64 len)
```

Comment:
I have discussed with Bruno. Let's use strong hash (SHA256) for assembly and files. Otherwise, we are open ourselves to attack scenarios. Specifically, let's use strong hash for BlobKind.AssemblyManagedDLL, BlobKind.AssemblyUnmanagedDir, BlobKind.JobDependencyFile